addons/mail/static/src/components/message_reply/message_reply.js

```diff
@@ -0,0 +1,32 @@
+/** @odoo-module */
+import { registerMessagingComponent } from '@mail/utils/messaging_component';
+
+const { Component } = owl;
+export class MessageReply extends Component {
+
+    /**
+     * @returns {mail.message}
+     */
+    get message() {
+        return this.messaging && this.messaging.models['mail.message'].get(this.props.messageLocalId);
+    }
+
+    highlightReply() {
+        const replyMessageComponent = this.env.messageList.messageRefs.find(
+            comp => comp.props.messageLocalId === this.message.localId
+        );
+        if (replyMessageComponent) {
+            replyMessageComponent.highlight();
+            replyMessageComponent.scrollIntoView({ behavior: 'smooth', block: 'center' });
+        }
+    }
```

Comment:
We don't want business logic at component level. User interactions should be dispatched to models ASAP, and this should change the state of the app in the models. Components should only reflect UI from the modelling in models.

So here, `highlightReply` should be dispatched to model, e.g. to a method of `Message` or `ThreadView` (the highlight looks contextual to the thread being viewed, so looks more appropriate putting state in `ThreadView` rather than `Message`).
The state of a message being highlighted should not be in component but in model, so as a field of message.

Scroll into view should be an adjust from model. This is a dynamic change, and to make it as declarative as possible, we use special things called "hints", which are created in models and components adjust UI based on these hints. Scrolling in threadview/message list is managed with "component hint" (see `addComponentHint` method on `ThreadView`). Focus is mostly done with `doFocus`/`isFocused` fields. The pattern is as follow:

- Models set `doX` fields / add component hints
- Components make UI adjustments based on these hints, then update models to clear the hints

Doing so put most logic in model and makes it much more declarative as it is. We want to reduce amount and length of algorithms as much as possible, and the trick to do this correctly is by expressing the logic with as much (meaningful) state as possible.

(btw, the state in `Message` component is a youthful mistake. We still keep it due to model `Message` and component `Message` being 2 different concepts, and we don't have "Message View" in model yet. This will come sooner or later, as we plan to turn everything into Model and Fields, so that everything is as declarative as you can get, even for the UI part of Discuss).